crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -488,12 +488,26 @@ def services
         end
 
         begin
-          unless horizon_node.nil?
+          unless horizon_node.nil? || monasca_node.nil?
             horizon_node.wait_for_script_to_finish(
               "/usr/sbin/crowbar-dump-monasca-db.sh",
               timeouts[:dump_grafana_db]
             )
           end
+        rescue StandardError => e
+          ::Crowbar::UpgradeStatus.new.end_step(
+            false,
+            services: {
+              data: "Problem while dumping Grafana database: " + e.message,
+              help: "Check /var/log/crowbar/production.log at admin server, " \
```

Comment:
"at" admin server but "on" horizon node :o)

Reply:
Fixed (and in a couple of other places, too. Let's break the bad grammar convention :-))